src/main/scala/com/exini/dicom/data/Elements.scala

```diff
@@ -788,8 +788,16 @@ case class Elements(characterSets: CharacterSets, zoneOffset: ZoneOffset, data:
   def toElements(withPreamble: Boolean = true): List[Element] =
     if (withPreamble) PreambleElement :: toList.flatMap(_.toElements) else toList.flatMap(_.toElements)
   def toParts(withPreamble: Boolean = true): List[DicomPart] = toElements(withPreamble).flatMap(_.toParts)
-  def toBytes(withPreamble: Boolean = true): ByteString =
-    data.map(_.toBytes).foldLeft(if (withPreamble) PreambleElement.toBytes else ByteString.empty)(_ ++ _)
+  def toBytes(withPreamble: Boolean = true): ByteString = {
+    val preambleBytes = if (withPreamble) PreambleElement.toBytes else ByteString.empty
+    val (fmiElements, dataElements) = data.partition(e => isFileMetaInformation(e.tag))
+    val fmiBytes = fmiElements.map(_.toBytes).foldLeft(ByteString.empty)(_ ++ _)
+    val dataBytes = dataElements.map(_.toBytes).foldLeft(ByteString.empty)(_ ++ _)
+    if (getString(Tag.TransferSyntaxUID).contains(UID.DeflatedExplicitVRLittleEndian))
```

Comment:
Your new utility function `isDeflated` could be used here? https://github.com/exini/dicom-streams/pull/25/files#diff-dc319249e595080f3e91f8a435f50638c73c4b79a93322bb37d99ee1faa91b52R80

Reply:
Good suggestion. Fixed